src/hotspot/cpu/aarch64/aarch64_vector.ad

```diff
@@ -3447,32 +3447,44 @@ instruct reduce_addHF(vRegF dst, vRegF fsrc, vReg vsrc, vReg tmp) %{
 // 1. Non strictly-ordered AddReductionVF when vector size > 128-bits. For example -
 //    AddReductionVF generated by Vector API. For vector size > 128-bits, it is more
 //    beneficial performance-wise to generate direct SVE instruction even if it is
 //    strictly ordered.
 // 2. Strictly-ordered AddReductionVF. For example - AddReductionVF generated by
 //    auto-vectorization on SVE machine.
-// For half float input (inferred as a T_SHORT type) - the only case that can reach here is
-// through autovectorization which is strictly ordered. At the time of writing this, support for
-// Float16 in VectorAPI is not available and thus only support for autovectorization has been added.
-instruct reduce_addFHF_sve(vRegF dst_src1, vReg src2) %{
-  predicate((Matcher::vector_element_basic_type(n->in(2)) == T_FLOAT &&
-             (!VM_Version::use_neon_for_vector(Matcher::vector_length_in_bytes(n->in(2))) ||
-              n->as_Reduction()->requires_strict_order())) ||
-            (Matcher::vector_element_basic_type(n->in(2)) == T_SHORT && UseSVE > 0));
-  match(Set dst_src1 (AddReductionVHF dst_src1 src2));
+instruct reduce_addF_sve(vRegF dst_src1, vReg src2) %{
+  predicate(!VM_Version::use_neon_for_vector(Matcher::vector_length_in_bytes(n->in(2))) ||
+            n->as_Reduction()->requires_strict_order());
   match(Set dst_src1 (AddReductionVF dst_src1 src2));
-  format %{ "reduce_addFHF_sve $dst_src1, $dst_src1, $src2" %}
+  format %{ "reduce_addF_sve $dst_src1, $dst_src1, $src2" %}
   ins_encode %{
     assert(UseSVE > 0, "must be sve");
     uint length_in_bytes = Matcher::vector_length_in_bytes(this, $src2);
     assert(length_in_bytes == MaxVectorSize, "invalid vector length");
-    BasicType bt = Matcher::vector_element_basic_type(this, $src2);
-    __ sve_fadda($dst_src1$$FloatRegister, __ elemType_to_regVariant(bt), ptrue, $src2$$FloatRegister);
+    __ sve_fadda($dst_src1$$FloatRegister, __ S, ptrue, $src2$$FloatRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+// This rule calculates the reduction result in strict order. Two cases will
+// reach here:
+// 1. Non strictly-ordered AddReductionVHF when vector size > 128-bits. For example -
+//    AddReductionVHF generated by Vector API. For vector size > 128-bits, it is more
+//    beneficial performance-wise to generate direct SVE instruction even if it is
+//    strictly ordered.
+// 2. Strictly-ordered AddReductionVHF. For example - AddReductionVHF generated by
+//    auto-vectorization on SVE machine.
+instruct reduce_addHF_sve(vRegF dst_src1, vReg src2) %{
+  predicate(UseSVE > 0);
+  match(Set dst_src1 (AddReductionVHF dst_src1 src2));
+  format %{ "reduce_addHF_sve $dst_src1, $dst_src1, $src2" %}
+  ins_encode %{
+    uint length_in_bytes = Matcher::vector_length_in_bytes(this, $src2);
+    assert(length_in_bytes == MaxVectorSize, "invalid vector length");
+    __ sve_fadda($dst_src1$$FloatRegister, __ H, ptrue, $src2$$FloatRegister);
   %}
   ins_pipe(pipe_slow);
 %}
```

Comment:
Could you please float this rule above `reduce_addF_sve` and below `reduce_addHF`? Better to rename `reduce_addHF` to `reduce_addHF_neon` ?